applets/grep/grep.go

```diff
@@ -59,7 +59,7 @@ func doGrep(pattern *regexp.Regexp, fh io.Reader, fn string, print_fn bool) {
 			return
 		}
 		if line == "" {
-			break
+			continue
```

Comment:
Although you are right that bailing on empty lines is indeed a bug, this is not the whole fix. Now the only way to exit the for-loop is by encountering an error, which in turn would make grep print an error message when encountering `io.EOF`. Could you add a `if` for that?